Utils/Pieces/King.py

```diff
@@ -9,37 +9,37 @@ def __init__(self, position,color):
     def can_move(self,board):
         moves = []
         attack_moves = []
-        castle_moves = self.castle(board.map)
+        castle_moves = self.castle(board)
 
         for i in (-1, 0, 1):
             for j in (-1, 0, 1):
                 new_x = self.x + i
                 new_y = self.y + j
                 if (i != 0 or j != 0) and new_x >= 0 and new_x < 8 and new_y >= 0 and new_y < 8:
-                    if board.map[new_x][new_y] is None:
-                        moves.append([new_x, new_y])
+                    if board.board[new_y][new_x] is None:
+                        moves.append([new_y,new_x])
                     else:
-                        if board.map[new_x][new_y].color != self.color:
-                            attack_moves.append([new_x, new_y])
+                        if board.board[new_y][new_x].color != self.color:
+                            attack_moves.append([new_y ,new_x])
 
         moves += castle_moves
         return moves, attack_moves
     def can_castle(self,rook):
         return rook is not None and rook.type == "Rook" and not rook.has_moved
-    def castle(self,map):
+    def castle(self,board):
         castle_moves = []
-        left_rook = map[0][self.y]
-        right_rook = map[7][self.y]
+        left_rook = board.board[self.y][0]
```

Comment:
That is not looking good... `board.board`? I would overwrite some `__get__` or something. Not use it like that.